src/spoke/VaultSpoke.sol

```diff
@@ -0,0 +1,382 @@
+    require(
+      SignatureChecker.isValidSignatureNow(params.depositor, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.depositor, params.nonce);
+    return
+      _executeMint({depositor: params.depositor, receiver: params.receiver, shares: params.shares});
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function withdrawWithSig(
+    EIP712Types.VaultWithdraw calldata params,
+    bytes calldata signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+    return
+      _executeWithdraw({
+        caller: msg.sender,
+        receiver: params.receiver,
+        owner: params.owner,
+        assets: params.assets
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function redeemWithSig(
+    EIP712Types.VaultRedeem calldata params,
+    bytes memory signature
+  ) public returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+    return
+      _executeRedeem({
+        caller: msg.sender,
+        receiver: params.receiver,
+        owner: params.owner,
+        shares: params.shares
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function depositWithPermit(
+    uint256 assets,
+    address receiver,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external returns (uint256) {
+    try
+      IERC20Permit(asset()).permit({
+        owner: msg.sender, // deposit only mints for caller
+        spender: address(this),
+        value: assets,
+        deadline: deadline,
+        v: v,
+        r: r,
+        s: s
+      })
+    {} catch {}
+    return deposit(assets, receiver);
+  }
+
+  /// @inheritdoc IERC20Permit
+  function permit(
+    address owner,
+    address spender,
+    uint256 value,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external {
+    require(block.timestamp <= deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(
+      keccak256(
+        abi.encode(
+          EIP712Hash.PERMIT_TYPEHASH,
+          owner,
+          spender,
+          value,
+          _useNonce({owner: owner, key: _PERMIT_NONCE_KEY}),
+          deadline
+        )
+      )
+    );
+    require(owner == ECDSA.recover({hash: digest, v: v, r: r, s: s}), InvalidSignature());
+    _approve({owner: owner, spender: spender, value: value});
+  }
+
+  /// @inheritdoc IERC4626
+  function previewDeposit(uint256 assets) public view virtual returns (uint256) {
+    return hub().previewAddByAssets(assetId(), assets);
+  }
```

Comment:
review-note: [ATokenVault](https://github.com/aave/Aave-Vault/blob/main/src/ATokenVault.sol#L384) min's this with maxDeposit but that is omitted here to more accurate with the spec

<img width="1091" height="338" alt="Image" src="https://github.com/user-attachments/assets/8b832b8b-a526-48ea-b447-4af95efc232d" />